File: go/dsa/algorithms/sliding-window/max-points.go
func maxScore(cardPoints []int, k int) int {
    n := len(cardPoints)
    windowSize := n - k
    totalSum := 0
    for i := 0; i < n; i++ {
        totalSum += cardPoints[i]
    }

    left := 0
    currentSum := 0
    minSum := 1 << 31 - 1 // Max int value

    for right := 0; right < n; right++ {
        // 1️⃣ Expand
        currentSum += cardPoints[right]

        // 2️⃣ When window size reaches k (n-k in this case)
        if right - left + 1 == windowSize {
            // 3️⃣ Record / Evaluate
            if currentSum < minSum {
                minSum = currentSum
            }

            // 4️⃣ Slide window
            currentSum -= cardPoints[left]
            left++
        }
    }
    if n == k { return totalSum }
    return totalSum - minSum
}

File: cpp/dsa/algorithms/sliding-window/max-points.cpp
int maxScore(vector<int>& cardPoints, int k) {
    int n = cardPoints.size();
    int windowSize = n - k;
    int totalSum = 0;
    for (int i = 0; i < n; i++) {
        totalSum += cardPoints[i];
    }

    int left = 0;
    int currentSum = 0;
    int minSum = numeric_limits<int>::max();

    for (int right = 0; right < n; right++) {
        // 1️⃣ Expand
        currentSum += cardPoints[right];

        // 2️⃣ When window size reaches k (n-k in this case)
        if (right - left + 1 == windowSize) {
            // 3️⃣ Record / Evaluate
            if (currentSum < minSum) {
                minSum = currentSum;
            }

            // 4️⃣ Slide window
            currentSum -= cardPoints[left];
            left++;
        }
    }
    if (n == k) return totalSum;
    return totalSum - minSum;
}